nixos/modules/services/torrent/synapse-bt.nix

```diff
@@ -0,0 +1,115 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.synapse-bt;
+in {
+  options = {
+    services.synapse-bt = {
+      enable = mkEnableOption "the Synapse BitTorrent daemon";
+
+      installClient = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Add the synapse-bt package, which includes a client,
+          <command>sycli</command>, to
+          <option>environment.systemPackages</option>.
+        '';
+      };
+
+      user = mkOption {
```

Comment:
As someone who has never used this program I would like to know if this program stores any data which would make `DynamicUser` an inappropriate choice for this module?

Reply:
If it was just the config, this is definitely possible, but I'm not exactly sure how DynamicUser deals with having data outside its directory (which is needed for downloads, DynamicUser is completely private iirc). For example, I'm using a custom downloads directory and `cfg.group` on my server with this module, so it can share its folder with other programs.

Reply:
In the case that this program has a downloads directory the user may access `DynamicUser` is not a good fit. Thanks for explaining.